resources/kube-controller-manager.yaml

```diff
@@ -12,7 +12,6 @@ spec:
       command:
         - kube-controller-manager
         ${cloud_config == "" ? "" : "- --cloud-config=/etc/kubernetes/config/cloud_provider/cloud.conf"}
-        ${cloud_provider == "" ? "" : "- --cloud-provider=${cloud_provider}"}
```

Comment:
I do not think we are doing this correctly. Reading through this: https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/ I think that there are 2 steps for kube-controller-manager to migrate to external cloud providers.
- First, you just roll the flag `--enable-leader-migration` to kube-controller-manager pods of kube version 1.26.
- Then, you change `--cloud-provider` to `external` as you roll to the latest kube and have already deployed the external cloud-controller.

Reply:
That does sound much better - rollout the external providers on 1.26 and then roll cleanly onto 1.27

Reply:
not quite. Reading again and again:

> The control plane now contains nodes of both version N and N + 1. The nodes of version N run kube-controller-manager only, and these of version N + 1 run both kube-controller-manager and cloud-controller-manager. The migrated controllers, as specified in the configuration, are running under either kube-controller-manager of version N or cloud-controller-manager of version N + 1 depending on which controller manager holds the migration lease. No controller will ever be running under both controller managers at any time.

I think that the cloud-controller has to come with the update and kube-controller-manager will know to not compete for the lease during the migration.